java/ql/src/Telemetry/ExternalAPI.qll

```diff
@@ -0,0 +1,52 @@
+/** Provides classes and predicates related to handling APIs from external libraries. */
+
+private import java
+private import APIUsage
+private import semmle.code.java.dataflow.ExternalFlow
+
+/**
+ * An external API from either the Java Standard Library or a 3rd party library.
+ */
+class ExternalAPI extends Callable {
+  ExternalAPI() { not this.fromSource() }
+
+  /** Holds true if this API is part of a common testing library or framework */
+  predicate isTestLibrary() { getDeclaringType() instanceof TestLibrary }
+
+  /**
+   * Gets information about the external API in the form expected by the CSV modeling framework.
+   */
+  string asCSV(ExternalAPI api) {
+    result =
+      api.getDeclaringType().getPackage() + ";?;" + api.getDeclaringType().getSourceDeclaration() +
```

Comment:
Minor: I think I prefer the CSV snippet without the `?`.
```suggestion
      api.getDeclaringType().getPackage() + ";;" + api.getDeclaringType().getSourceDeclaration() +
```

Reply:
This has been a discussion with @yo-h as well. Initially we thought we'd align the format with our CSV model for easy copy&paste when modeling. But given other efforts (e.g. generating models), we started to lean towards using a more user-friendly and readable format like `java.lang.Object#Object()`. Any thoughts on this?

Reply:
Fine by me - I just found the `?` a bit awkward.